yt_dlp/extractor/crunchyroll.py

```diff
@@ -827,22 +893,76 @@ def _real_extract(self, url):
     }, {
         'url': 'https://beta.crunchyroll.com/it/series/GY19NQ2QR/Girl-Friend-BETA',
         'only_matching': True,
     }]
 
     def _real_extract(self, url):
-        lang, series_id = self._match_valid_url(url).group('lang', 'id')
-        return self.url_result(f'https://www.crunchyroll.com/{lang}{series_id.lower()}',
-                               CrunchyrollShowPlaylistIE.ie_key(), series_id)
+        lang, internal_id, display_id = self._match_valid_url(url).group('lang', 'id', 'display_id')
+
+        if not self._get_cookies(url).get('etp_rt'):
+            initial_state, app_config = self._get_beta_embedded_json(self._download_webpage(url, display_id), display_id)
+            series_id = re.sub(r'-{2,}', '-', initial_state['content']['byId'][internal_id]['slug_title'])
+            self.to_screen(f'{display_id}: Not logged in. Switching extractors. New url: https://www.crunchyroll.com/{lang}{series_id}')
+            return self.url_result(f'https://www.crunchyroll.com/{lang}{series_id}',
+                                   CrunchyrollShowPlaylistIE.ie_key(), series_id)
+
+        api_domain, bucket, params = self._get_params(lang)
+
+        series_response = self._download_json(
+            f'{api_domain}/cms/v2{bucket}/series/{internal_id}', display_id,
+            note='Retrieving series metadata',
+            query=params)
+
+        seasons_response = self._download_json(
+            f'{api_domain}/cms/v2{bucket}/seasons?series_id={internal_id}', display_id,
+            note='Retrieving season list',
+            query=params)
+
+        def entries():
+            for season in seasons_response['items']:
+                season_id = season['id']
+                episodes_response = self._download_json(
+                    f'{api_domain}/cms/v2{bucket}/episodes?season_id={season_id}', display_id,
+                    note='Retrieving episode list for %s' % season.get('slug_title'),
+                    query=params)
+                for episode in episodes_response['items']:
+                    episode_id = episode['id']
+                    episode_display_id = episode['slug_title']
+                    yield self.url_result(
+                        f'https://beta.crunchyroll.com/{lang}watch/{episode_id}/{episode_display_id}',
+                        CrunchyrollBetaIE.ie_key(), episode_id, episode['title'],
+                        description=episode.get('description').replace(r'\r\n', '\n'),
+                        duration=float_or_none(episode.get('duration_ms'), 1000),
+                        series=episode.get('series_title'),
+                        series_id=episode.get('series_id'),
+                        season=episode.get('season_title'),
+                        season_id=episode.get('season_id'),
+                        season_number=episode.get('season_number'),
+                        episode=episode.get('title'),
+                        episode_number=episode.get('sequence_number'))
```

Comment:
There is nothing wrong with doing it this way, so I'm not gonna force you to change this. But it looks really ugly imho
```py
                    yield {
                        '_type': 'url',
                        'url': f'https://beta.crunchyroll.com/{lang}watch/{episode_id}/{episode_display_id}',
                        'ie_key': CrunchyrollBetaIE.ie_key(),
                        'id': episode_id,
                        ...
                    }
```

Reply:
Ok, fair point. I'll convert it.